scripts/add_elevation_to_network.py

```diff
@@ -63,36 +77,50 @@
     elevation_dictionary = n.get_node_elevation_dictionary(elevation_tif_file_path=elevation, null_value=tif_null_value)
 
     if save_dict_to_json is True:
-        with open(os.path.join(elevation_output_dir, 'elevation/node_elevation_dictionary.json'), 'w',
+        with open(os.path.join(elevation_output_dir, 'node_elevation_dictionary.json'), 'w',
                   encoding='utf-8') as f:
             json.dump(sanitiser.sanitise_dictionary(elevation_dictionary), f, ensure_ascii=False, indent=4)
 
-    logging.info('Validating the elevation data added to network nodes')
-    report = elevation.validation_report_for_node_elevation(elevation_dictionary)
+
+    logging.info('Validating the node elevation data')
+    report = genet.utils.elevation.validation_report_for_node_elevation(elevation_dictionary)
     logging.info(report['summary'])
 
     if save_dict_to_json is True:
-        with open(os.path.join(elevation_output_dir, 'elevation/validation_report_for_elevation.json'), 'w',
+        with open(os.path.join(elevation_output_dir, 'validation_report_for_elevation.json'), 'w',
                   encoding='utf-8') as f:
             json.dump(sanitiser.sanitise_dictionary(report), f, ensure_ascii=False, indent=4)
 
+
+    if write_elevation_to_network is True:
+        logging.info('Adding node elevation as attribute to the network')
+        node_attrib_dict = {}
+        for node_id in elevation_dictionary.keys():
+            elevation_value = elevation_dictionary[node_id]['z']
+            node_attrib_dict[node_id] = {'z': elevation_value}
+
+        n.apply_attributes_to_nodes(node_attrib_dict)
+
+
     logging.info('Creating slope dictionary for network links')
     slope_dictionary = n.get_link_slope_dictionary(elevation_dict=elevation_dictionary)
 
     if save_dict_to_json is True:
-        with open(os.path.join(elevation_output_dir, 'elevation/link_slope_dictionary.json'), 'w',
+        with open(os.path.join(elevation_output_dir, 'link_slope_dictionary.json'), 'w',
                   encoding='utf-8') as f:
-            json.dump(sanitiser.sanitise_dictionary(report), f, ensure_ascii=False, indent=4)
+            json.dump(sanitiser.sanitise_dictionary(slope_dictionary), f, ensure_ascii=False, indent=4)
+
 
-    logging.info('Adding link slope as attribute to the network')
+    if write_slope_to_network is True:
```

Comment:
you can just do: `if write_slope_to_network` since `write_slope_to_network` is a bool